core/controllers/resources_test.py

```diff
@@ -62,6 +62,30 @@ def test_image_upload_and_download(self):
         self.assertEqual(response.content_type, 'image/png')
         self.assertEqual(response.body, raw_image)
 
+    def test_unexpected_extensions_are_accepted(self):
```

Comment:
Can we add some comments here explaining what's going on (i.e. explain that imghdr has false negatives)? It would be nice if we could actually add that false-negative image to core/tests so that we have a test case available in the future, but if it's not freely-licensed then adding a link to it should be fine (assuming it's available on the web).